frontend/src/components/WikiList.js

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+
+class WikiList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleWikiChange = this.handleWikiChange.bind(this)
+    this.state = {error: false, wikis: []}
+  }
+
+  handleWikiChange(wikiId) {
+    console.log("Clicked wiki " + wikiId)
+    this.props.handleWikiChange(wikiId)
+  }
+
+  componentDidMount() {
+    fetch("http://localhost:8000/api/w", { crossDomain: true })
+      .then((res) => res.json())
+      .then(
+        (returnedWikis) => {
+            if (Array.isArray(returnedWikis)) {
+              this.setState({wikis: returnedWikis, error: null});
+            } else {
+              this.setState({wikis: [], error: "Invalid response"})
+            }
+        },
+        (e) => {
+          this.setState({wikis: [], error: e});
+        }
+      );
```

Comment:
This works, but `async`/`await` is a more popular setup. Here's an early sketch:

```js
try {
  const res = await fetch("http://localhost:8000/api/w", { crossDomain: true });
  const wikis = await res.json();
  if (!Array.isArray(wikis)) {
    throw new Error("Invalid response");
  }
  this.setState({ wikis, error: null });
} catch (error) {
  this.setState({ wikis: [], error });
}
```

You might see a lint error complaining that `componentDidMount` doesn't want you to use an `async` function. This is a good error, because you might call `this.setState` after the component is unmounted. For example, imagine the following sequence of events:

1. The component mounts
2. `componentDidMount` is called, firing off an HTTP request
3. The component is unmounted (e.g., because the user navigates away)
4. The HTTP request finishes, updating some state on a now-unmounted component

This is one of the big benefits of hooks: it's easier (IMO) to avoid this kind of issue. I'd switch this whole component to use hooks, put this HTTP request in a `useEffect`, and then ignore the response if the component is unmounted. Something like this:

```js
// This is an untested sketch
const [wikis, setWikis] = useState([]);
const [error, setError] = useState(null);

useEffect(() => {
  let isCanceled = false;

  (async () => {
    let newWikis = [];
    let newError = null;

    try {
      const res = await fetch("http://localhost:8000/api/w", {
        crossDomain: true,
      });

      // As an optimization, don't read the response body if we've canceled.
      if (isCanceled) {
        return;
      }

      if (Array.isArray(wikis)) {
        newWikis = await res.json();
      } else {
        newError = new Error("Invalid response");
      }
    } catch (error) {
      newError = error;
    }

    if (isCanceled) {
      return;
    }

    setWikis(newWikis);
    setError(newError);
  })();

  return () => {
    isCanceled = true;
  };
}, []);
```

There are a lot of different ways to write this (including using a higher-level helper like [SWR](https://swr.vercel.app/)) but that's how I'd approach it.